`fourmolu` linker failure on `x86_64-linux`

On 1a013a2858f96587223dbe2c626ff6e400167857, we get the following linker failure when building the latest `fourmolu` in `tools` on `x86_64-linux`:

```
Preprocessing executable 'fourmolu' for fourmolu-0.6.0.0..
--
  | Building executable 'fourmolu' for fourmolu-0.6.0.0..
  | [1 of 2] Compiling Paths_fourmolu   ( dist/build/fourmolu/autogen/Paths_fourmolu.hs, dist/build/fourmolu/fourmolu-tmp/Paths_fourmolu.o, dist/build/fourmolu/fourmolu-tmp/Paths_fourmolu.dyn_o )
  | [2 of 2] Compiling Main             ( app/Main.hs, dist/build/fourmolu/fourmolu-tmp/Main.o, dist/build/fourmolu/fourmolu-tmp/Main.dyn_o )
  | Linking dist/build/fourmolu/fourmolu ...
  | /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: error: /nix/store/3m5k46qh6jv5shfr2sby5s9v1l3lbjm5-ghc-9.2.2/lib/ghc-9.2.2/rts/libHSrts.a(Globals.o): multiple definition of 'ghc_unique_inc'
  | /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: /nix/store/njgcfnjq187g9kcayajpb7wdl6mdf39i-ghc-lib-parser-lib-ghc-lib-parser-9.2.3.20220527/lib/x86_64-linux-ghc-9.2.2/ghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC/libHSghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC.a(genSym.o): previous definition here
  | /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: error: /nix/store/3m5k46qh6jv5shfr2sby5s9v1l3lbjm5-ghc-9.2.2/lib/ghc-9.2.2/rts/libHSrts.a(Globals.o): multiple definition of 'ghc_unique_counter'
  | /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: /nix/store/njgcfnjq187g9kcayajpb7wdl6mdf39i-ghc-lib-parser-lib-ghc-lib-parser-9.2.3.20220527/lib/x86_64-linux-ghc-9.2.2/ghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC/libHSghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC.a(genSym.o): previous definition here
  | collect2: error: ld returned 1 exit status
  | `gcc' failed in phase `Linker'. (Exit code: 1)
  | error: builder for '/nix/store/4vg3bz0jwgjbis9zp4qxw2azjrdn0m5w-fourmolu-exe-fourmolu-0.6.0.0.drv' failed with exit code 1;
  | last 10 log lines:
  | > Building executable 'fourmolu' for fourmolu-0.6.0.0..
  | > [1 of 2] Compiling Paths_fourmolu   ( dist/build/fourmolu/autogen/Paths_fourmolu.hs, dist/build/fourmolu/fourmolu-tmp/Paths_fourmolu.o, dist/build/fourmolu/fourmolu-tmp/Paths_fourmolu.dyn_o )
  | > [2 of 2] Compiling Main             ( app/Main.hs, dist/build/fourmolu/fourmolu-tmp/Main.o, dist/build/fourmolu/fourmolu-tmp/Main.dyn_o )
  | > Linking dist/build/fourmolu/fourmolu ...
  | > /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: error: /nix/store/3m5k46qh6jv5shfr2sby5s9v1l3lbjm5-ghc-9.2.2/lib/ghc-9.2.2/rts/libHSrts.a(Globals.o): multiple definition of 'ghc_unique_inc'
  | > /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: /nix/store/njgcfnjq187g9kcayajpb7wdl6mdf39i-ghc-lib-parser-lib-ghc-lib-parser-9.2.3.20220527/lib/x86_64-linux-ghc-9.2.2/ghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC/libHSghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC.a(genSym.o): previous definition here
  | > /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: error: /nix/store/3m5k46qh6jv5shfr2sby5s9v1l3lbjm5-ghc-9.2.2/lib/ghc-9.2.2/rts/libHSrts.a(Globals.o): multiple definition of 'ghc_unique_counter'
  | > /nix/store/x2zbyl2l4xayv4d5is0b31plfq8xjjym-binutils-2.35.2/bin/ld.gold: /nix/store/njgcfnjq187g9kcayajpb7wdl6mdf39i-ghc-lib-parser-lib-ghc-lib-parser-9.2.3.20220527/lib/x86_64-linux-ghc-9.2.2/ghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC/libHSghc-lib-parser-9.2.3.20220527-GvRaPbyuMD52PUlH6TgbMC.a(genSym.o): previous definition here
  | > collect2: error: ld returned 1 exit status
  | > `gcc' failed in phase `Linker'. (Exit code: 1)
  | For full logs, run 'nix log /nix/store/4vg3bz0jwgjbis9zp4qxw2azjrdn0m5w-fourmolu-exe-fourmolu-0.6.0.0.drv'.
  | error: 1 dependencies of derivation '/nix/store/f9wbr7qnv5zimaypkc9sz00jdi94wlka-pre-commit-config.json.drv' failed to build
  | error: 1 dependencies of derivation '/nix/store/294rnn5cdnawgkacx35phfkmqqbwpyrr-pre-commit-run.drv' failed to build
  | 🚨 Error: The command exited with status 1
  | user command error: exit status 1


```

(This is new in the last week; previously `fourmolu` was building fine with the workaround in #1337.)

--- a/modules/configuration-nix.nix
+++ b/modules/configuration-nix.nix
@@ -58,7 +58,8 @@ in {
   packages.mintty.components.library.build-tools = pkgs.lib.mkForce [];
 
   packages.ghc-lib-parser.patches = [
-    (fromUntil "8.10.0.0" "9.1" ../overlays/patches/ghc-lib-parser-8.10-global-unique-counters-in-rts.patch)
+    (fromUntil "8.10.0.0" "9.2" ../overlays/patches/ghc-lib-parser-8.10-global-unique-counters-in-rts.patch)
+    (fromUntil "9.2.0.0" "9.3" ../overlays/patches/ghc-lib-parser-9.2-global-unique-counters-in-rts.patch)
   ];
 
   # See https://github.com/haskell-nix/hnix/pull/1053

--- a/overlays/patches/ghc-lib-parser-9.2-global-unique-counters-in-rts.patch
+++ b/overlays/patches/ghc-lib-parser-9.2-global-unique-counters-in-rts.patch
@@ -1,0 +1,20 @@
+diff --git a/compiler/cbits/genSym.c b/compiler/cbits/genSym.c
+index ecb318d5a8..8e70db33fc 100644
+--- a/compiler/cbits/genSym.c
++++ b/compiler/cbits/genSym.c
+@@ -1,9 +1,15 @@
+ #include <Rts.h>
+ #include <assert.h>
+ #include "Unique.h"
++#include "ghcversion.h"
+
++// These global variables have been moved into the RTS.  It allows them to be
++// shared with plugins even if two different instances of the GHC library are
++// loaded at the same time (#19940)
++#if !MIN_VERSION_GLASGOW_HASKELL(9,2,0,0)
+ HsInt ghc_unique_counter = 0;
+ HsInt ghc_unique_inc     = 1;
++#endif
+
+ #define UNIQUE_BITS (sizeof (HsInt) * 8 - UNIQUE_TAG_BITS)
+ #define UNIQUE_MASK ((1ULL << UNIQUE_BITS) - 1)

